test/e2e/operator_test.go

```diff
@@ -152,6 +158,89 @@ func TestOperatorAvailable(t *testing.T) {
+	}
+
+	t.Log("Creating external dns instance with source type route")
+	extDNS := helper.buildOpenShiftExternalDNS(testExtDNSName, hostedZoneID, hostedZoneDomain)
+	if err := kubeClient.Create(context.TODO(), &extDNS); err != nil {
+		t.Fatalf("Failed to create external DNS %q: %v", testExtDNSName, err)
+	}
+	defer kubeClient.Delete(context.TODO(), &extDNS)
+
+	// create a service of type ClusterIP without annotation
+	t.Log("Creating test route service")
+	service := testRouteService(testServiceName, testNamespace)
+	if err := kubeClient.Create(context.Background(), service); err != nil {
+		t.Fatalf("Failed to create test route service %s/%s: %v", testNamespace, testServiceName, err)
+	}
+	defer kubeClient.Delete(context.TODO(), service)
+
+	// create a route with the annotation targeted by the ExternalDNS resource
+	t.Log("Creating source route")
+	route := testRoute(testRouteName, testNamespace, testRouteHost, testServiceName)
+	if err := kubeClient.Create(context.Background(), route); err != nil {
+		t.Fatalf("Failed to create test route %s/%s: %v", testNamespace, testRouteName, err)
+	}
+	defer kubeClient.Delete(context.TODO(), route)
+
+	// get the route canonical name
+	var targetRoute routev1.Route
+	err = kubeClient.Get(context.TODO(), types.NamespacedName{
+		Namespace: testNamespace,
+		Name:      testRouteName,
+	}, &targetRoute)
+	if err != nil {
+		t.Fatalf("Failed to retrieve the created route %s/%s: %v", testNamespace, testRouteName, err)
+	}
+	targetRouteCName := targetRoute.Status.Ingress[0].RouterCanonicalHostname
+	if targetRouteCName == "" {
+		t.Fatalf("Created route's canonical name is empty %v", err)
+	}
+	// get the route IPs by looking up the route canonical name
+	routeIPs := make(map[string]struct{})
+	routeExpectedIPs, err := customResolver("").LookupIP(context.TODO(), "ip", targetRouteCName)
+	if err != nil {
+		t.Fatalf("Failed to retrieve the IPs from the created route canonical name %s: %v", targetRouteCName, err)
+	}
+
+	// create lookup for assertion
+	for _, ip := range routeExpectedIPs {
+		routeIPs[ip.String()] = struct{}{}
+	}
+
+	// try all nameservers and fail only if all failed
+	for _, nameSrv := range nameServers {
+		t.Logf("Looking for DNS record in nameserver: %s", nameSrv)
+		// create a DNS resolver which uses the nameservers of the test hosted zone
+		customResolver := customResolver(nameSrv)
+
+		// verify dns records has been created for the route host.
+		if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
+			actualIPs, err := customResolver.LookupHost(context.TODO(), testRouteHost)
+			if err != nil {
+				t.Logf("Waiting for dns record: %s", testRouteHost)
+				return false, nil
+			}
+			for _, ip := range actualIPs {
+				if _, ok := routeIPs[ip]; !ok {
+					return false, nil
+				}
+			}
+			return true, nil
+		}); err != nil {
+			t.Logf("Failed to verify that DNS has been correctly set.")
+		} else {
+			return
+		}
+	}
```

Comment:
This part seems to be repeated 1-1 from the service test case. Maybe it's worth putting this into `utils.go`.